internal/migration/precheck.go

```diff
@@ -71,6 +74,20 @@ func SourcePrecheck(
 	return nil
 }
 
+// ImportPrecheck checks the data being imported to make sure preconditions for
+// importing are met.
+func ImportPrecheck(
```

Comment:
I do wonder if we should revisit how all this is plumbed together. For now, this is fine.

Reply:
It seems like it would be well worth doing. It would be good to at least move the migration checks that are currently in the upgradevalidation package to here, i.e. move them from the source controller to the target controller. A lot of them use state but it should be possible to just look at the actual model description.